db_stress_tool/db_stress_common.h

```diff
@@ -327,16 +330,97 @@ extern inline std::string Key(int64_t val) {
   return big_endian_key;
 }
 
+// A struct for maintaining the parameters for generating variable length keys
+struct KeyGenContext {
+  // Max length of the key in 8-byte units. Each level is a int64 encoded in
+  // big endian format
+  unsigned int levels;
+  // Number of keys of each possible length in a given window
+  std::vector<uint64_t> weights;
+  // Size of window for varying the key length
+  uint64_t window;
```

Comment:
It took me a long time to figure out what window meant here. It's really the number of adjacent keys in a cycle of key lengths.